include/rocksdb/status.h

```diff
@@ -18,31 +18,16 @@
 
 #include <string>
 #include "rocksdb/slice.h"
+#ifdef OS_WIN
+#include <string.h>
+#endif
+#include <cstring>
 
 namespace rocksdb {
 
-class Status {
+template <class T>
```

Comment:
RocksDB code base has very limited use of template (which follows the guideline of Google C++ Style https://google.github.io/styleguide/cppguide.html#Template_metaprogramming ), and I don't remember any public interface uses template. This makes the public interface harder to understand. I suggest we think twice before using template here.

Reply:
Fair enough. I wanted to reuse the code/subcodes and template seemed to be the way to do that. But on further reflection, it allows us to reuse the values but the names have to be qualified by classname. Maybe we can use ```Status::Code``` and ```Status::SubCode``` directly in ```IOStatus```? That way, we can have a single set of error codes, but have a distinct set of named constructors in ```IOStatus``` to indicate what's allowable.